src/main/java/ws/wamp/jawampa/WampRouter.java

```diff
@@ -675,32 +701,70 @@ private void onMessageFromRegisteredChannel(WampRouterHandler handler, WampMessa
+            for (Subscription prefixSubscription : prefixSubscriptionMap.values()) {
+                if (pub.topic.startsWith(prefixSubscription.topic)) {
+                    publishEvent(handler, pub, publicationId, prefixSubscription);
+                }
+            }
+
+            Map<String, Subscription> wildcardSubscriptionMap = handler.realm.subscriptionsByType.get(Subscription.Type.wildcard);
+            String[] components = pub.topic.split("\\.", -1);
+            for (Subscription wildcardSubscription : wildcardSubscriptionMap.values()) {
+                boolean matched = true;
+                if (components.length == wildcardSubscription.components.length) {
+                    for (int i=0; i < components.length; i++) {
+                        if (wildcardSubscription.components[i].length() > 0
+                                && !components[i].equals(wildcardSubscription.components[i])) {
+                            matched = false;
+                            break;
                         }
-                        if (skipPublisher) continue;
                     }
-                    // Publish the event to the subscriber
-                    EventMessage ev = new EventMessage(subscription.subscriptionId, publicationId,
-                        null, pub.arguments, pub.argumentsKw);
-                    receiver.ctx.writeAndFlush(ev);
+                }else
+                    matched = false;
+
+                if (matched) {
+                    publishEvent(handler, pub, publicationId, wildcardSubscription);
                 }
             }
-            
+
             if (sendAcknowledge) {
                 PublishedMessage response = new PublishedMessage(pub.requestId, publicationId);
                 handler.ctx.writeAndFlush(response);
             }
         }
     }
+
+    private void publishEvent(WampRouterHandler publisher, PublishMessage pub, long publicationId, Subscription subscription){
+        for (WampRouterHandler receiver : subscription.subscribers) {
+            if (receiver == publisher ) { // Potentially skip the publisher
+                boolean skipPublisher = true;
+                if (pub.options != null) {
+                    JsonNode excludeMeNode = pub.options.get("exclude_me");
+                    if (excludeMeNode != null) {
+                        skipPublisher = excludeMeNode.asBoolean(true);
+                    }
+                }
+                if (skipPublisher) continue;
+            }
+
+            ObjectNode details = null;
+            if (subscription.type != Subscription.Type.exact) {
+                details = new ObjectMapper().createObjectNode();
+                details.put("topic", pub.topic);
+            }
+
+            // Publish the event to the subscriber
+            EventMessage ev = new EventMessage(subscription.subscriptionId, publicationId,
```

Comment:
I think this was a leftover because before my last change for each subscriber a new subscription Id was generated and therefore this was necessary. With the current mechanism changing it to a single event which is sent out to all subscribers is OK.